client/src/components/advancedSelectWidget/MultiTypeAdvancedSelectComponent.js

```diff
@@ -113,61 +114,101 @@ const widgetTypeMapping = {
     setEntityType(newEntityType)
     setAdvancedSelectProps(widgetTypeMapping[newEntityType])
   }
-  let searchPlaceholder = ""
+  const searchPlaceholder = useMemo(() => {
+    const [key] = Object.entries(ENTITY_TYPES).find(
+      ([, et]) => et === entityType
+    )
+    const entityLabel = SEARCH_OBJECT_LABELS[SEARCH_OBJECT_TYPES[key]]
+    return "Find " + entityLabel.toLowerCase()
+  }, [entityType])
   return (
     <>
-      <ButtonToggleGroup value={entityType} onChange={changeEntityType}>
-        {Object.entries(ENTITY_TYPES).map((key, value) => {
-          const entityName = key[1]
-          const entityLabel = SEARCH_OBJECT_LABELS[SEARCH_OBJECT_TYPES[key[0]]]
-          if (entityName === entityType) {
-            searchPlaceholder = "Find " + entityLabel.toLowerCase()
-          }
-          return (
-            <Button key={entityName} value={entityName}>
-              {entityLabel}
-            </Button>
-          )
-        })}
-      </ButtonToggleGroup>
+      {entityTypes.length > 1 && (
+        <ButtonToggleGroup value={entityType} onChange={changeEntityType}>
+          {Object.entries(ENTITY_TYPES)
+            .filter(([, et]) => entityTypes.includes(et))
+            .map(([key, entityName], value) => {
+              const entityLabel = SEARCH_OBJECT_LABELS[SEARCH_OBJECT_TYPES[key]]
+              return (
+                <Button key={entityName} value={entityName}>
+                  {entityLabel}
+                </Button>
+              )
+            })}
+        </ButtonToggleGroup>
+      )}
 
-      <AdvancedSingleSelect
-        autofocus="true"
-        fieldName="entitySelect"
-        fieldLabel="Search in ANET:"
-        placeholder={searchPlaceholder}
-        value={{}}
-        showEmbedded
-        overlayColumns={advancedSelectProps.overlayColumns}
-        overlayRenderRow={advancedSelectProps.overlayRenderRow}
-        filterDefs={advancedSelectProps.filterDefs}
-        onChange={value => onConfirm(value, entityType)}
-        objectType={advancedSelectProps.objectType}
-        queryParams={advancedSelectProps.queryParams}
-        fields={advancedSelectProps.fields}
-        addon={advancedSelectProps.addon}
-      />
+      {isMultiSelect ? (
+        <AdvancedMultiSelect
+          autofocus="true"
+          fieldName={fieldName}
+          fieldLabel="Search in ANET:"
+          placeholder={searchPlaceholder}
+          value={value}
+          showEmbedded
+          overlayColumns={advancedSelectProps.overlayColumns}
+          overlayRenderRow={advancedSelectProps.overlayRenderRow}
+          filterDefs={advancedSelectProps.filterDefs}
+          onChange={value => onConfirm(value, entityType)}
+          objectType={advancedSelectProps.objectType}
+          queryParams={advancedSelectProps.queryParams}
+          fields={advancedSelectProps.fields}
+          addon={advancedSelectProps.addon}
+        />
+      ) : (
+        <AdvancedSingleSelect
```

Comment:
The props of the 2 components are almost the same and it's a huge list. Can't we do something like:
```
const SelectComponent = isMultiSelect ? AdvancedMultiSelect : AdvancedSingleSelect
const extraSelectProps = isMultiSelect ? { } : { showRemoveButton: false} 
```
and than use them like
```
<SelectComponent
  ... all those props
  {...extraSelectprops}
/>
```

Reply:
Indeed.